### DISCUSSION TITLE
integration for body scale: beurer bf720

### DISCUSSION THREAD

**Discussion by usrstephan** [theengs/decoder#521] [New devices/things integration]
I need some help:
after enabling Advertisement and Advanced Data I got the following message:
{"id":"E2:DB:xx:xx:xx:xx","mac_type":0,"adv_type":0,"name":"BF720","manufacturerdata":"11060001","rssi":-66,"txpower":0,"servicedata":"0102","servicedatauuid":"0x181d"}.
I would really like to add some integration (https://community.openmqttgateway.com/t/integration-for-bathroom-scales-possible/3145/2) but I am not able to follow the docs.
What can I do next? Is there some software for linux to get the further data needed?

**Comment by DigiH** [comment]
Hi @usrstephan 

This is really all that is needed to see if a decoder for your scale will be possible.

`{"id":"E2:DB:xx:xx:xx:xx","mac_type":0,"adv_type":0,"name":"BF720","manufacturerdata":"11060001","rssi":-66,"txpower":0,"servicedata":"0102","servicedatauuid":"0x181d"}`

With MQTT messages like this could you also post the weight the scale showed at the very same time, and I suppose the impedance as well, which is possibly hidden behind the BMI etc. calculations - so let's start with the weight to see if and where it might be encoded in the broadcast **manufacturerdata** or **servicedata**.

**Reply by usrstephan** [reply]
so sorry, the rssi is changing (mqtt explorer), but both manufacturerdata and servicedata stay exactly the same (66.3kg). Even together with a chair (77kg) - no change. Will try it again tomorrow.
Any idea how to get more information?

**Reply by DigiH** [reply]
These freely broadcast advertising **manufacturerdata** and **servicedata** are the only way for Theengs Decoder to currently have decoders for devices which broadcast useful decodable data  like this.

If the **manufacturerdata** or **servicedata** do not change with different weights it looks like your Beurer scale is not compatible with Theengs Decoder as it is now.

Best to verify with a few more weighings.

In that case it might be possible to [READ a certain service/characteristic with OpenMQTTGateway](https://docs.openmqttgateway.com/use/ble.html#read-write-ble-characteristics-over-mqtt) from the scale, for which you would need some smart phone app like nRFConnect to see which services and characteristics the scale has when connecting to it, and which might hold the information you want to read.

**Comment by usrstephan** [comment]
checked it again several times - no changes.
sorry for asking further details - I am a little bit confused:
within the app I find the scale:
![Bildschirmfoto vom 2024-03-01 14-59-55](https://github.com/theengs/decoder/assets/38238708/f487bda8-5ab6-4514-9f1b-dd14f6a59425)
battery level shows up without any problems
![Bildschirmfoto vom 2024-03-01 15-00-19](https://github.com/theengs/decoder/assets/38238708/38261c61-dcb4-409a-b0a1-d062d13c8f8d)
but I do not find information about weight
![Bildschirmfoto vom 2024-03-01 15-00-38](https://github.com/theengs/decoder/assets/38238708/8055b05c-172f-44c4-8895-f8c69f804eed)
do you know how to get more details or is it possible to define the needed command for openmqttgateway with the information provided?

**Reply by DigiH** [reply]
Theengs Decoder can currently only decoder freely broadcast advertising data, so a decoder is not possible for the Beurer  BF720.

With your results from nRFConnect however you can see which service/chars might possibly give you the information you want.

Using the OpenMQTTGateway's READ command for example with service/char 180F/2A19 will give you the battery level for example, but this READing will have to be added through your controller.

**Comment by usrstephan** [comment]
thanks for clarifying
battery level reading seems clear but I do not see any option to get the weight although I checked all services in the app.
Could there be further options to find the service/char for that?

**Reply by DigiH** [reply]
Have a look through the SIG Assigned Numbers PDF to see if your scale adheres to the standard weight scale/weight service/chars.

https://www.bluetooth.com/specifications/assigned-numbers/
